akka-stream/src/main/scala/akka/stream/impl/io/IOSources.scala

```diff
@@ -20,7 +20,7 @@ import scala.concurrent.{ Future, Promise }
 private[akka] final class SynchronousFileSource(f: File, chunkSize: Int, val attributes: Attributes, shape: SourceShape[ByteString])
   extends SourceModule[ByteString, Future[Long]](shape) {
   override def create(context: MaterializationContext) = {
-    val mat = ActorMaterializer.downcast(context.materializer)
+    val mat = ActorMaterializer.downcast(context.materializer) //FIXME why does this need to rely on a concrete Materializer impl?
```

Comment:
for that line below:

```
     val ref = mat.actorOf(context, props.withDispatcher(dispatcher))
```

It would be possible to make it actor-less for the synchronous file source, I don't think we need to do that for 1.0 in any case, in general we could I guess if it's worth it, yet I'd rather not implement actor-less for asynchronous-io once we get to it though.